lib/go/Makefile

```diff
@@ -15,15 +15,37 @@ endif
 export GOPATH
 
 
+################################################################################
+##                                   DEP                                      ##
+################################################################################
+
+DEP ?= ./dep
+DEP_VER ?= 0.3.1
+DEP_BIN := dep-$$GOHOSTOS-$$GOHOSTARCH
+DEP_URL := https://github.com/golang/dep/releases/download/v$(DEP_VER)/$$DEP_BIN
+
+$(DEP):
+	GOVERSION=$$(go version | awk '{print $$4}') && \
+	GOHOSTOS=$$(echo $$GOVERSION | awk -F/ '{print $$1}') && \
+	GOHOSTARCH=$$(echo $$GOVERSION | awk -F/ '{print $$2}') && \
+	DEP_BIN="$(DEP_BIN)" && \
+	DEP_URL="$(DEP_URL)" && \
+	curl -sSLO $$DEP_URL && \
+	chmod 0755 "$$DEP_BIN" && \
+	mv -f "$$DEP_BIN" "$@"
+ifneq (./dep,$(DEP))
+dep: $(DEP)
```

Comment:
I'm not sure I follow. This would expand to

```
dep: ./dep
```

And if `DEP=dep` is given, this will expand to
```
dep: dep
```
which will likely break.

I wonder if we shouldn't do

```
DEP ?= dep
...
.PHONY: $(DEP)
$(DEP):
    ...

ineq (dep, $(DEP))
.PHONY: dep
dep: $(DEP)
endif

vendor: dep
...
```

And then, looking at that, I wonder how valuable it is to have DEP configurable?

Reply:
If we drop the configurability this could be

```
... remove 'DEP ?= ./dep' line ... 

dep:
    ... current contents of $(DEP)

vendor: dep
    dep ...

```

Reply:
This is really nit-picky and I'm more than happy if this is merged as is and we revisit this with a use case if it breaks at some point.

Reply:
The point is to allow you to specify an absolute path to an existing dep binary. You would never specify just "dep" as it would be the same binary that gets downloaded.

Reply:
SGTM